### PR TITLE
Fix Slider not moving in short distances.

### PR DESCRIPTION
**Pull request by bernaferrari**
## Description

`Slider` currently relies in `HorizontalDragGestureRecognizer` which currently relies in `kTouchSlop` to detect if user tapped or scrolled. Issue is, you always want a Slider to be scrollable. It makes no sense for a Slider that is in position 0.3 to only begin scrolling in position 0.4. This also causes a bad user experience (as per the GIF below).
I decided to modify `HorizontalDragGestureRecognizer` and `VerticalDragGestureRecognizer` to receive an optional touchSlop parameter which defaults to `kTouchSlop`, so that there is no behaviour or breaking change. This way, any user or widget might specify their own touchSlop value, which is what I did with the `Slider`. I am open to discuss other approaches or modifying my implementation.

## Related Issues

Screen Recording 2019-09-11 at 23 09 28 2019-09-11 23_12_13

https://github.com/flutter/flutter/issues/40310

## Tests

How do I even test this?

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

### CODE REVIEW

**Comment by bernaferrari** [comment]
Could someone please take a look?

**Comment by SwissCheese5** [comment]
This is definitely an issue that this of fixes. I've seen it happen multiple times as well as the stuttery or stopped or clipped motion while actively changing the slider value automatically as a video plays.

**Comment by Hixie** [comment]
This should only be visible if you can also scroll vertically. If there's no competing vertical drag gesture (or some other competing gesture, e.g. a higher-level tap gesture detector), it should start dragging immediately.

Unless I'm misunderstanding the issue?

File: packages/flutter/lib/src/material/slider.dart [hunk 1]
```diff
@@ -672,7 +672,7 @@ class _RenderSlider extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
        _textDirection = textDirection {
     _updateLabelPainter();
     final GestureArenaTeam team = GestureArenaTeam();
-    _drag = HorizontalDragGestureRecognizer()
+    _drag = HorizontalDragGestureRecognizer(touchSlop: 0)
```
Comment (Afsar-Pasha) [inline packages/flutter/lib/src/material/slider.dart]: As @Hixie mentioned `touchSlop` should be `> 0` something like `8` or `kTouchSlop/2` to avoid changing slider when scrolling page, list, etc
Also make this change in `CupertinoSlider`.

Comment (bernaferrari) [reply packages/flutter/lib/src/material/slider.dart]: As I mentioned in the issue,

In flutter_gallery, the range sliders are inside a Column. There is no scrolling factor, yet this triggers the issue.

Additionally, regarding the slop, even in ListView, I've never seen this in other platforms like Android or iOS. Shouldn't it be optional?

Comment (Afsar-Pasha) [reply packages/flutter/lib/src/material/slider.dart]: @Hixie, if Android and iOS have this behavior, shouldn't this be the default behavior of Material`Slider` and `CupertinoSlider`?

Comment (bernaferrari) [reply packages/flutter/lib/src/material/slider.dart]: GIF-191213_131837

Android settings app

Comment (Hixie) [reply packages/flutter/lib/src/material/slider.dart]: When I test Android here it behaves as I would expect Flutter to behave (there's a slop in both the horizontal and vertical directions).

But that's not the issue here. The issue is that this used to work, and then it regressed (according to the issue). We need to bisect to find when it regressed. The best way to do that is to create a unit test that exactly reproduces the issue you're concerned with, verify that it worked in the past, and then use `git bisect` to automatically determine what commit broke it.

**Comment by Hixie** [comment]
I'm going to close this PR because before we can determine if it is the right fix we need to figure out what broke.

File: packages/flutter/lib/src/material/slider.dart (hunk 1 above)
Comment (clocksmith) [reply packages/flutter/lib/src/material/slider.dart]: What is the expected behavior of the horizontal slider when it is in a container that has horizontal scroll (such as a TabView)?

Before the RangeSlider was added to the Gallery (https://github.com/flutter/flutter/commit/04bd87794d0704096b70308f7ecb52f1a92c62b1), the Slider demo was not in a TabView, and moved on short distances. Once the RangeSlider was added, a TabView was added for Slider and RangeSlider demo to be on separate tabs, and then the issues started surfacing.

In other words, this issue seemed to have always existed, but was only surfaced in the gallery once the Slider demo was put into a container with horizontal scroll.
